### PR TITLE
Multitask metrics support

### PR DESCRIPTION
**Pull request by rbharath**
Adds support for multitask metrics:

1. A multitask metric is defined as a metric that has a `task_averager` that averages computed metrics across tasks.
2. The `HyperparamOpt` test now supports optimization of multitask metrics.

### CODE REVIEW

**Comment by rbharath** [comment]
@joegomes This PR fixes the issue you found with `HyperparamOpt` and keras models (using the fix you suggested). You should be able to update to master once this is merged and see the issue go away.
